core/src/main/java/org/elasticsearch/monitor/os/OsProbe.java

```diff
@@ -19,12 +19,198 @@
+
+        try {
+            method = osMxBean.getClass().getDeclaredMethod("getSystemLoadAverage");
+            method.setAccessible(true);
+
+        } catch (Exception e) {
+            // not available
+
+        }
+        getSystemLoadAverage = method;
+    }
+
+    /**
+     * Returns the amount of free physical memory in bytes.
+     */
+    public static long getFreePhysicalMemorySize() {
+        if (getFreePhysicalMemorySize == null) {
+            return -1;
+
+        }
+        try {
+            return (Long) getFreePhysicalMemorySize.invoke(osMxBean);
+
+        } catch (Exception e) {
+            return -1;
+
+        }
+    }
+
+    /**
+     * Returns the total amount of physical memory in bytes.
+     */
+    public static long getTotalPhysicalMemorySize() {
+        if (getTotalPhysicalMemorySize == null) {
+            return -1;
+
+        }
+        try {
+            return (Long) getTotalPhysicalMemorySize.invoke(osMxBean);
+
+        } catch (Exception e) {
+            return -1;
+
+        }
+    }
+
+    /**
+     * Returns the amount of free swap space in bytes.
+     */
+    public static long getFreeSwapSpaceSize() {
+        if (getFreeSwapSpaceSize == null) {
+            return -1;
+
+        }
+        try {
+            return (Long) getFreeSwapSpaceSize.invoke(osMxBean);
+
+        } catch (Exception e) {
+            return -1;
+
+        }
+    }
+
+    /**
+     * Returns the total amount of swap space in bytes.
+     */
+    public static long getTotalSwapSpaceSize() {
+        if (getTotalSwapSpaceSize == null) {
+            return -1;
+
+        }
+        try {
+            return (Long) getTotalSwapSpaceSize.invoke(osMxBean);
+
+        } catch (Exception e) {
+            return -1;
+
+        }
+    }
+
+    /**
+     * Returns the system load average for the last minute.
+     */
+    public static double getSystemLoadAverage() {
+        if (getSystemLoadAverage == null) {
+            return -1;
+
+        }
+        try {
+            return (Long) getSystemLoadAverage.invoke(osMxBean);
+
+        } catch (Exception e) {
+            return -1;
+
+        }
+    }
+
+    @Inject
+    public OsProbe(Settings settings) {
+        super(settings);
+    }
+
+    public OsInfo osInfo() {
+        OsInfo info = new OsInfo();
+        info.availableProcessors = Runtime.getRuntime().availableProcessors();
+        info.name = Constants.OS_NAME;
+        return info;
+    }
+
+    public OsStats osStats() {
+        OsStats stats = new OsStats();
+        stats.timestamp = System.currentTimeMillis();
+        stats.loadAverage = getSystemLoadAverage();
+
+        OsStats.Mem mem = new OsStats.Mem();
+        mem.total = getTotalPhysicalMemorySize();
+        mem.free = getFreePhysicalMemorySize();
+        mem.freePercent = calculatePercentage(mem.free, mem.total);
+        mem.used = mem.total - mem.free;
+        mem.usedPercent = calculatePercentage(mem.used, mem.total);
```

Comment:
maybe we can remove the `freePercent`, `used`, `usedPercent` as fields, and add getters for them that compute it on the fly?